DataSetExplorer/DataSetBuilder/Model/DataSetAnnotation.cs

```diff
@@ -8,36 +8,36 @@ public class DataSetAnnotation
         public CodeSmell InstanceSmell { get; }
         public int Severity { get; }
         public List<SmellHeuristic> ApplicableHeuristics { get; }
-        public int AnnotatorId { get; }
+        public Annotator Annotator { get; set; }
 
-        public DataSetAnnotation(string instanceSmell, int severity, int annotatorId, List<SmellHeuristic> applicableHeuristics)
+        public DataSetAnnotation(string instanceSmell, int severity, Annotator annotator, List<SmellHeuristic> applicableHeuristics)
         {
             InstanceSmell = new CodeSmell(instanceSmell);
             Severity = severity;
-            AnnotatorId = annotatorId;
+            Annotator = annotator;
             ApplicableHeuristics = applicableHeuristics;
             Validate();
         }
 
         private void Validate()
         {
             if (Severity < 0 || Severity > 3) throw new InvalidOperationException("Accepted severity ranges from 0 to 3, but was " + Severity);
-            if (AnnotatorId == 0) throw new InvalidOperationException("Annotator ID is required.");
-            if (Severity > 0 && ApplicableHeuristics.Count < 1) throw new InvalidOperationException("Annotations made by " + AnnotatorId + " with severity " + Severity + " must have at least one applicable heuristic.");
+            if (Annotator.Id == 0) throw new InvalidOperationException("Annotator ID is required.");
+            if (Severity > 0 && ApplicableHeuristics.Count < 1) throw new InvalidOperationException("Annotations made by " + Annotator.Id + " with severity " + Severity + " must have at least one applicable heuristic.");
         }
 
-        public override int GetHashCode() => (AnnotatorId, InstanceSmell: InstanceSmell.Value).GetHashCode();
+        public override int GetHashCode() => (Annotator.Id, InstanceSmell: InstanceSmell.Value).GetHashCode();
 
         public override bool Equals(object other)
         {
             return other is DataSetAnnotation annotation
-                   && AnnotatorId.Equals(annotation.AnnotatorId)
+                   && Annotator.Id.Equals(annotation.Annotator.Id)
```

Comment:
This can be moved to the annotator class, so that we have Annotator.Equals(Annotator) here. Don't forget the hash code (this might help https://docs.microsoft.com/en-us/visualstudio/ide/reference/generate-equals-gethashcode-methods?view=vs-2019 or find a similar article).